### ISSUE TITLE
Resizing bounding boxes after inference does not work with transforms.v2.Resize

### ISSUE THREAD

**Issue by Rayndell** [pytorch/vision#9315]
### 🐛 Describe the bug

I trained a detection model producing bounding boxes with labels and scores. What I want after that is export the model to ONNX, but I want to incorporate as much as possible of the preprocessing and postprocessing operations directly into the model to make its use as easy as possible.

In particular, my images are not of the same size and a resize operation is included in the preprocessing functions. After inference the boxes' coordinates are relative to the resized image size, and I want to resize them to fit the original size. But I am having troubles to do so with the transforms.v2.Resize function as it seems to not do anything. Below is my code:

```python
class DetectionModel(nn.Module):
    def __init__(self, base_model, resize_height, resize_width, nms_threshold=0.6):
        super(DetectionModel, self).__init__()
        self.base_model = base_model
        self.nms_threshold = nms_threshold
        self.resize_height = resize_height
        self.resize_width = resize_width
        self.transform = v2.Compose([
            v2.ToImage(),
            v2.Resize(size=(resize_height, resize_width), antialias=False),
            v2.ToDtype(torch.float32, scale=True)
        ])
        
    def forward(self, input):
        img_height, img_width = input.shape[-2:]
        resize = v2.Resize(size=(img_height, img_width), antialias=False)
        
        # Apply the preprocessing steps
        input = self.transform(input)
        
        # Apply the base model to the preprocessed input
        output = self.base_model(input)
        
        all_boxes = []
        all_scores = []
        all_labels = []
        
        for i in range(len(output)):
            pred = output[i]
            boxes = pred['boxes']
            scores = pred['scores']
            labels = pred['labels']

            # Non max suppression
            keep = ops.nms(boxes, scores, iou_threshold=self.nms_threshold)
            boxes = boxes[keep]
            scores = scores[keep]
            labels = labels[keep]
            
            # Replacing the boxes coordinates in the original image before resizing
            boxes = BoundingBoxes(boxes, format="XYXY", canvas_size=(img_height, img_width))
            boxes = resize(boxes)
            
            all_boxes.append(boxes)
            all_scores.append(scores)
            all_labels.append(labels)
        
        return torch.stack(all_boxes), torch.stack(all_scores), torch.stack(all_labels)
    
    def export_to_onnx(self, path, device):
        self.eval()
        input_names = ["input"]
        output_names = ["boxes", "scores", "labels"]
        dynamic_axes = {
            "input": {0: "batch_size", 2: "height", 3: "width"},
            "scores": {0: "batch_size", 1: "num_detections"},
            "labels": {0: "batch_size", 1: "num_detections"},
            "boxes": {0: "batch_size", 1: "num_detections"}
        }

        input_example = torch.randint(0, 255, (1, 3, 1024, 1024), dtype=torch.uint8).to(device)
        torch.onnx.export(
            self,
            input_example,
            path,
            export_params=True,
            opset_version=16,
            do_constant_folding=True,
            input_names=input_names,
            output_names=output_names,
            dynamic_axes=dynamic_axes,
            dynamo=False
        )
```

Interestingly, I found a workaround by doing this:

```python
sx = torch.tensor(img_width / float(self.resize_width),  dtype=boxes.dtype, device=boxes.device)
sy = torch.tensor(img_height / float(self.resize_height), dtype=boxes.dtype, device=boxes.device)

x_idx = torch.tensor([0, 2], device=boxes.device)
y_idx = torch.tensor([1, 3], device=boxes.device)
boxes[:, x_idx] = boxes[:, x_idx] * sx
boxes[:, y_idx] = boxes[:, y_idx] * sy
```

The problem is when my number of detection boxes is 0 it crashes. I tried to make a test condition with boxes.numel() > 0 but I've got the following warning message when exporting, and it never passes the test in the exported model:

TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!

What should I do? Is there a bug in the Resize function (it is supposed to work with bounding boxes)?

Any help will be greatly appreciated.

### Versions

Collecting environment information...
PyTorch version: 2.9.1+cu126
Is debug build: False
CUDA used to build PyTorch: 12.6
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Professionnel (10.0.26100 64 bits)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: N/A

Python version: 3.12.10 (tags/v3.12.10:0cc8128, Apr  8 2025, 12:21:36) [MSC v.1943 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26100-SP0
Is CUDA available: True
CUDA runtime version: 12.6.20
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1080 Ti
Nvidia driver version: 581.80
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\bin\cudnn_ops64_9.dll
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Name: Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz
Manufacturer: GenuineIntel
Family: 198
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 3696
MaxClockSpeed: 3696
L2CacheSize: 1536
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] onnx==1.17.0
[pip3] onnx-ir==0.1.12
[pip3] onnxruntime==1.23.2
[pip3] onnxruntime-gpu==1.23.2
[pip3] onnxscript==0.5.6
[pip3] optree==0.16.0
[pip3] segmentation_models_pytorch==0.5.0
[pip3] tf2onnx==1.16.1
[pip3] torch==2.9.1+cu126
[pip3] torchaudio==2.9.1+cu126
[pip3] torchmetrics==1.8.2
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.24.1+cu126
[conda] Could not collect

**Comment by gaurav-redhat** [comment]
###  BoundingBoxes Resize Fix

## Problem
`v2.Resize` does nothing when `canvas_size` equals the resize target.

## Fix
```python
# ❌ Wrong: canvas_size = target size (no scaling)
boxes = BoundingBoxes(boxes, format="XYXY", canvas_size=(img_height, img_width))

# ✅ Correct: canvas_size = current coordinate space
boxes = BoundingBoxes(boxes, format="XYXY", canvas_size=(self.resize_height, self.resize_width))
```

**Comment by Rayndell** [comment]
Hi, thank you very much for your answer. But it did not solve my problem, since I encountered a strange behavior of the Resize function. Actually, when the input of my model is an image the same size as the resize size, here is the output when I do not resize anything, which is good:

<img width="1022" height="759" alt="Image" src="https://github.com/user-attachments/assets/95e181be-3269-4175-9919-2ce2e3e12b4a" />

But when I activate the Resize function in the exported onnx model (using the instruction you provided), here is the output:

<img width="1022" height="763" alt="Image" src="https://github.com/user-attachments/assets/c4c9ded9-16b5-40bb-9792-4563f285330f" />

It seems that the x coordinates are ok but the y coordinates are stretched, but in theory the resizing is not supposed to do anyhing in this case since the image size is identical to the resize size.

**Comment by gaurav-redhat** [comment]
For ONNX Export with Empty Boxes
For the ONNX export issue with zero detections, instead of a Python boolean check, use torch.where or ensure padding

```
def _scale_boxes(self, boxes, orig_height, orig_width):
     # Manual scaling - guaranteed correct behavior
    sx = orig_width / self.resize_width
    sy = orig_height / self.resize_height
    
    # boxes shape: [N, 4] in XYXY format
    scale = torch.tensor([sx, sy, sx, sy], dtype=boxes.dtype, device=boxes.device)
    return boxes * scale

```

**Comment by Rayndell** [comment]
All right, great news, I have been able to fix the problem and the Resize function works as intended. What is strange is what I have modified to get to this result.

In fact, the problem was the size of the first input I gave to the model prior to the ONNX export:

`input_example = torch.randint(0, 255, (1, 3, 1024, 1024), dtype=torch.uint8).to(device)

torch.onnx.export(
            self,
            input_example,
            path,
            export_params=True,
            opset_version=16,
            do_constant_folding=True,
            input_names=input_names,
            output_names=output_names,
            dynamic_axes=dynamic_axes,
            dynamo=False
 )`

What I did is that I modified the input_example size from (1024, 1024) (which was taken as random at the time) to (768, 1024), which happens to be equal to the resize size. For some reason the model is taking the example input size given at the export as the input size during the inference, no matter the real input size. When the input example size is equal to the resize size this does not happen and the resize works as intended. I am unable to explain this, but it could be some king of bug.

**Comment by Rayndell** [comment]
Another thing is the apparition of these warnings during the inference:

2025-12-29 13:10:18.9474557 [W:onnxruntime:, execution_frame.cc:874 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {-1,-1} does not match actual shape of {1} for output /base_model/roi_heads/box_roi_pool/Squeeze_3_output_0
2025-12-29 13:10:18.9561536 [W:onnxruntime:, execution_frame.cc:874 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {-1,-1} does not match actual shape of {110} for output /base_model/roi_heads/box_roi_pool/Squeeze_2_output_0
2025-12-29 13:10:18.9765772 [W:onnxruntime:, execution_frame.cc:874 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {-1,-1} does not match actual shape of {350} for output /base_model/roi_heads/box_roi_pool/Squeeze_1_output_0
2025-12-29 13:10:19.0284572 [W:onnxruntime:, execution_frame.cc:874 onnxruntime::ExecutionFrame::VerifyOutputSizes] Expected shape from model of {-1,-1} does not match actual shape of {539} for output /base_model/roi_heads/box_roi_pool/Squeeze_output_0

Is this something to worry about?
